examples/index.html

```diff
@@ -1,121 +1,118 @@
-      rel="stylesheet"
-    />
-    <link
-      rel="stylesheet"
-      href="../node_modules/leaflet/dist/leaflet.css"
-      type="text/css"
-      media="screen"
-      title="no title"
-      charset="utf-8"
-    />
-    <script src="../node_modules/leaflet-toolbar/dist/leaflet.toolbar.js"></script>
-    <link
-      href="../node_modules/leaflet-toolbar/dist/leaflet.toolbar.css"
-      rel="stylesheet"
-    />
+  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
 
-    <!-- for full-res export -->
-    <script src="../node_modules/jquery/dist/jquery.js"></script>
-    <script src="../node_modules/webgl-distort/dist/webgl-distort.js"></script>
-    <script src="../node_modules/glfx/glfx.js"></script>
+  <script src="../node_modules/leaflet/dist/leaflet.js" type="text/javascript" charset="utf-8"></script>
+  <link href="../node_modules/font-awesome/css/font-awesome.min.css" rel="stylesheet" />
+  <link rel="stylesheet" href="../node_modules/leaflet/dist/leaflet.css" type="text/css" media="screen" title="no title"
+    charset="utf-8" />
+  <script src="../node_modules/leaflet-toolbar/dist/leaflet.toolbar.js"></script>
+  <link href="../node_modules/leaflet-toolbar/dist/leaflet.toolbar.css" rel="stylesheet" />
+  <!-- for pattern-mining utility -->
+  <script src="../node_modules/matcher-core/orb.core.com.js"></script>
+  <script src="../src/util/matcher/init_with_matcher.js"></script>
+  <script src="../src/util/matcher/projector.js"></script>
+  <script src="../src/util/matcher/stitcher.js"></script>
+  <script src="../src/util/matcher/init_.js"></script>
+  <!-- for full-res export -->
+  <script src="../node_modules/jquery/dist/jquery.js"></script>
+  <script src="../node_modules/webgl-distort/dist/webgl-distort.js"></script>
+  <script src="../node_modules/glfx/glfx.js"></script>
 
-    <!-- for EXIF geocode -->
-    <script
-      type="text/javascript"
-      src="../node_modules/exif-js/exif.js"
-    ></script>
+  <!-- for EXIF geocode -->
+  <script type="text/javascript" src="../node_modules/exif-js/exif.js"></script>
 
-    <link
-      rel="stylesheet"
-      href="../dist/leaflet.distortableimage.css"
-      type="text/css"
-      media="screen"
-      title="no title"
-      charset="utf-8"
-    />
-    <script src="../dist/leaflet.distortableimage.js"></script>
-  </head>
-  <body style="margin:0;">
-    <form id="test_form">
-      <input type="file" id="inputimage" accept="image/*" />
-    </form>
+  <link rel="stylesheet" href="../dist/leaflet.distortableimage.css" type="text/css" media="screen" title="no title"
+    charset="utf-8" />
+  <script src="../dist/leaflet.distortableimage.js"></script>
+</head>
 
-    <div
-      id="map"
-      style="width:100%; height:100%; position:absolute; top:0;"
-    ></div>
-  </body>
-  <script>
-    var map;
+<body>
+  <center>
+    <div class="loader-container">
+      <img width="100" src="leaflet.gif">
```

Comment:
So, what is happening while this is showing? The matcher is searching for interest points in some images?

Reply:
Yup! But now with the caching implemented, I guess we won't be needing this at all from the second iteration onwards, and the initial iteration should take way lesser loading time (~7000 to ~15 frames perf improvement), so I guess it would be better if we should synchronously process those?